lib/livebook_web/plugs/auth_plug.ex

```diff
@@ -14,7 +14,14 @@ defmodule LivebookWeb.AuthPlug do
     mode = Livebook.Config.auth_mode()
 
     if authenticated?(conn, mode) do
-      conn
+      if redirect_to = get_session(conn, :redirect_to) do
```

Comment:
Actually, we should check for redirect in the auth controller immediately when we authenticate :D